packages/flutter/lib/src/cupertino/spell_check_suggestions_toolbar.dart

```diff
@@ -0,0 +1,131 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart' show SelectionChangedCause, SuggestionSpan;
+import 'package:flutter/widgets.dart';
+
+import 'text_selection_toolbar.dart';
+import 'text_selection_toolbar_button.dart';
+
+/// iOS only shows 3 spell check suggestions in the toolbar.
+const int _maxSuggestions = 3;
+
+/// The default spell check suggestions toolbar for iOS.
+///
+/// Tries to position itself below the [anchors], but if it doesn't fit, then it
+/// readjusts to fit above bottom view insets.
+class CupertinoSpellCheckSuggestionsToolbar extends StatelessWidget {
+  /// Constructs a [CupertinoSpellCheckSuggestionsToolbar].
+  const CupertinoSpellCheckSuggestionsToolbar({
+    super.key,
+    required this.anchors,
+    required this.buttonItems,
+  });
+
+  /// The location on which to anchor the menu.
+  final TextSelectionToolbarAnchors anchors;
+
+  /// The [ContextMenuButtonItem]s that will be turned into the correct button
+  /// widgets and displayed in the spell check suggestions toolbar.
+  ///
+  /// See also:
+  ///
+  ///  * [AdaptiveTextSelectionToolbar.buttonItems], the list of
+  ///    [ContextMenuButtonItem]s that are used to build the buttons of the
+  ///    text selection toolbar.
+  ///  * [SpellCheckSuggestionsToolbar.buttonItems], the list of
+  ///    [ContextMenuButtonItem]s used to build the Material style spell check
+  ///    suggestions toolbar.
+  final List<ContextMenuButtonItem> buttonItems;
+
+  /// Builds the button items for the toolbar based on the available
+  /// spell check suggestions.
+  static List<ContextMenuButtonItem>? buildButtonItems(
+    BuildContext context,
+    EditableTextState editableTextState,
+  ) {
+    // Determine if composing region is misspelled.
+    final SuggestionSpan? spanAtCursorIndex =
+      editableTextState.findSuggestionSpanAtCursorIndex(
+        editableTextState.currentTextEditingValue.selection.baseOffset,
+      );
+
+    if (spanAtCursorIndex == null) {
+      return null;
+    }
+    if (spanAtCursorIndex.suggestions.isEmpty) {
+      return <ContextMenuButtonItem>[
+        ContextMenuButtonItem(
+          onPressed: () {},
+          label: 'No Replacements Found',
+        )
+      ];
+    }
+
+    final List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+
+    // Build suggestion buttons.
+    int suggestionCount = 0;
+    for (final String suggestion in spanAtCursorIndex.suggestions) {
+      if (suggestionCount >= _maxSuggestions) {
+        break;
+      }
+      buttonItems.add(ContextMenuButtonItem(
+        onPressed: () {
+          if (!editableTextState.mounted) {
```

Comment:
@LongCatIsLooong like this? I think this is causing a test to break: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8786435562172940817/+/u/run_test.dart_for_web_tests_shard_and_subshard_6/test_stdout

Reply:
Yeah this is what I meant. Looks like the failure was caused by `Path.combine` not being implemented by the HTML renderer, but I double this mounted check is the direct cause of that failure.